odc/stats/plugins/gm.py

```diff
@@ -78,18 +78,25 @@ def native_transform(self, xx: xr.Dataset) -> xr.Dataset:
         return xx
 
     def input_data(self, datasets: Sequence[Dataset], geobox: GeoBox) -> xr.Dataset:
-        erased = load_enum_filtered(
-            datasets,
-            self._mask_band,
-            geobox,
-            categories=self.cloud_classes,
-            filters=self.filters,
-            groupby=self.group_by,
-            resampling=self.resampling,
-            chunks={},
-        )
+        cloud_shadow_mask = None
+        for cloud_class, filter in zip(self.cloud_classes, self.filters):
```

Comment:
I don't think it'll be working without changing the interface of the class.
-  `load_enum_filtered` requires a list of `str` or `int` as categories, and a list of `tuple` as filters. After this line, the input will be `str` or `int`, and `tuple` respectively. 

- In this change, it's assuming all the `cloud_classes` will be dealt separately. It was not `load_enum_filtered` intended for hence it's not what this input  intended for. We might want to get this very clear by separating two scenarios, a) smash all `cloud_classes` together to have one boolean, or b) keep them separate to have multiple booleans.

Reply:
The assumption here is that the input filter looks something like this:
`[[["opening", 0], ["dilation", 5]], [["opening", 0], ["dilation", 5]]]`>. 

This will be a change in the input.  I don't think we need to keep the single filter support at least for the summary products.   I'm happy to add the support if we think it's needed.

Reply:
I still need to update the type annotation for filters.

Reply:
so, if that's the case the `cloud_classes` will have the form of `[[a], [b]]` as well? it'll support the single filter by input being `[[a, b]]`.

Reply:
Alternatively I can add a separate input argument for separate filters to the gm plugin.  something that accepts a dictionary where keys are cloud classes and values are filters:
```
{'cloud': [['opening', 0], ['dilation', 5]],
 'shadow': [['opening', 0], ['dilation', 5]]}
```

Reply:
I like this one better actually.